src/wallet/test/psbt_wallet_tests.cpp

```diff
@@ -59,15 +59,14 @@ BOOST_AUTO_TEST_CASE(psbt_updater_test)
     const CTransaction txConst(*psbtx.tx);
 
     // Fill transaction with our data
-    TransactionError err;
     bool complete = true;
-    FillPSBT(&m_wallet, psbtx, err, complete, false, true);
+    BOOST_REQUIRE_EQUAL(TransactionError::OK, FillPSBT(&m_wallet, psbtx, complete, SIGHASH_ALL, false, true));
 
     // Get the final tx
     CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
     ssTx << psbtx;
     std::string final_hex = HexStr(ssTx);
-    BOOST_CHECK_EQUAL(final_hex, "70736274ff01009a020000000258e87a21b56daf0c23be8e7070456c336f7cbaa5c8757924f545887bb2abdd750000000000ffffffff838d0427d0ec650a68aa46bb0b098aea4422c071b2ca78352a077959d07cea1d0100000000ffffffff0270aaf00800000000160014d85c2b71d0060b09c9886aeb815e50991dda124d00e1f5050000000016001400aea9a2e5f0f876a588df5546e8742d1d87008f00000000000100bb0200000001aad73931018bd25f84ae400b68848be09db706eac2ac18298babee71ab656f8b0000000048473044022058f6fc7c6a33e1b31548d481c826c015bd30135aad42cd67790dab66d2ad243b02204a1ced2604c6735b6393e5b41691dd78b00f0c5942fb9f751856faa938157dba01feffffff0280f0fa020000000017a9140fb9463421696b82c833af241c78c17ddbde493487d0f20a270100000017a91429ca74f8a08f81999428185c97b5d852e4063f6187650000000104475221029583bf39ae0a609747ad199addd634fa6108559d6c5cd39b4c2183f1ab96e07f2102dab61ff49a14db6a7d02b0cd1fbb78fc4b18312b5b4e54dae4dba2fbfef536d752ae00000000");
+    BOOST_CHECK_EQUAL(final_hex, "70736274ff01009a020000000258e87a21b56daf0c23be8e7070456c336f7cbaa5c8757924f545887bb2abdd750000000000ffffffff838d0427d0ec650a68aa46bb0b098aea4422c071b2ca78352a077959d07cea1d0100000000ffffffff0270aaf00800000000160014d85c2b71d0060b09c9886aeb815e50991dda124d00e1f5050000000016001400aea9a2e5f0f876a588df5546e8742d1d87008f00000000000100bb0200000001aad73931018bd25f84ae400b68848be09db706eac2ac18298babee71ab656f8b0000000048473044022058f6fc7c6a33e1b31548d481c826c015bd30135aad42cd67790dab66d2ad243b02204a1ced2604c6735b6393e5b41691dd78b00f0c5942fb9f751856faa938157dba01feffffff0280f0fa020000000017a9140fb9463421696b82c833af241c78c17ddbde493487d0f20a270100000017a91429ca74f8a08f81999428185c97b5d852e4063f6187650000000104475221029583bf39ae0a609747ad199addd634fa6108559d6c5cd39b4c2183f1ab96e07f2102dab61ff49a14db6a7d02b0cd1fbb78fc4b18312b5b4e54dae4dba2fbfef536d752ae2206029583bf39ae0a609747ad199addd634fa6108559d6c5cd39b4c2183f1ab96e07f049c4942a9220602dab61ff49a14db6a7d02b0cd1fbb78fc4b18312b5b4e54dae4dba2fbfef536d704b9147fd300000000");
```

Comment:
A bit of context for why the hash value has been updated here.
If you look at how `FillPSBT` is called a few lines above, you may notice it got an extra `SIGHASH_ALL` parameter compared to the previous version.
It turns out this parameter has for some reason been missing from #4229 where this code has been backported.
The same code line in the original PR bitcoin#14978 contains the `SIGHASH_ALL` parameter as due.
In result, the next value, `false`, has been treated as a sighash type (implicitly converted to `0`) instead, which resulted in a different output value.
So here I'm restoring the original set of parameters.

CC @kittywhiskers: please let us know if the change in #4229 was intentional - my understanding is that this was a mistake, but maybe I'm missing something.

Reply:
Nice catch! 🥳 

Since we needed to regenerate our test vectors on account of not having SegWit, an error must've slipped past (otherwise tests would've pouted and we'd have to look into it but here we regenerated the vectors, that safety net's gone) 

It might be that I just woke up but I found myself searching `wallet/psbtwallet.cpp` instead of `wallet/test/psbt_wallet_tests.cpp`, took me a hot minute to it to click but consider doing something like this in the future:

> The [change made in 5229347@#4299](https://github.com/dashpay/dash/commit/522934703a7c23e39c064a04fd09586dbb146f12#diff-757244fb65c2fb785bcedb1be097be3252d516dd353c1cbd77ed50ae28f6603fR64) don't match the original  [change made in #14978](https://github.com/bitcoin/bitcoin/pull/14978/files#diff-757244fb65c2fb785bcedb1be097be3252d516dd353c1cbd77ed50ae28f6603fR66)

Saves you lines and pin-points the error ~~and helps groggy folks who just woke up comprehend better~~ 😉